nntrainer/src/databuffer_file.cpp

```diff
@@ -289,20 +289,19 @@ int DataBufferFromDataFile::setDataFile(std::string path, DataType type) {
   case DATA_VAL: {
     validation[type] = true;
     if (!data_file.good()) {
-      ml_logw("Warning: Cannot open validation data file. Cannot validate "
+      ml_loge("Error: Cannot open validation data file. Cannot validate "
               "training result");
       validation[type] = false;
-      break;
+      return ML_ERROR_INVALID_PARAMETER;
```

Comment:
I think the reason we set the break is that we still can run the training without validation data and test data set.

Reply:
Yes. `setSomething` should validates if given input is correct if not return error. It is the responsibility of caller how the error should treated.
e.g) `setValidationDataSet` should return error when it `validationDataSet` is not present. because it was not successful. 

This patch do as follows
1. if `dataset:validData` key is not present, it does not call `setDataFile` because it is not necessary
2. if `dataset:validData` key is present but path is invalid, `loadConfig` fails
3. if `dataset:trainData` key is not present, `loadConfig` fails

Reply:
Then, I think we need some kind of mode. We have to check at least train data is provided properly when Training mode. And also have to add run validation Mode only if no train data and validation data is provided without train.

Reply:
> Then, I think we need some kind of mode. We have to check at least train data is provided properly when Training mode. And also have to add run validation Mode only if no train data and validation data is provided without train. I mean we have to run the mode according to which data set is provided.

We can do this but that issue is not related to this PR. 

Setter still have to verify if current input is valid. and it is caller's responsibility whether to call this or not.  It is not related to whether is is required property or not.

In other words, It is expected behavior that if given input is not valid, setter return error.

For example, our capi expects to receive error when file is not valid and current implementation of `setDataFile` is actually causing bug to capi. ( eg) when `valid_file` is not found )

https://github.com/nnstreamer/nntrainer/blob/c16261fd34fe0cad582638fa62d4da5208d1b614/api/capi/src/nntrainer.cpp#L674-L714